functions.php

```diff
@@ -230,3 +230,22 @@ function redirect_private_pages_to_join_page() {
 }
 
 add_action( 'template_redirect', __NAMESPACE__ . '\\redirect_private_pages_to_join_page' );
+
+/**
+ * Add id attribute to <h1>, <h2>, <h3>, <h4>, <h5> tags
+ *
+ * @param string $content Content of the current post.
+ *
+ * @return string $content The filtered post content.
+ */
+function anchor_content_headings( $content ) {
+    $content = preg_replace_callback( "/\<h([1|2|3|4|5])\>(.*?)\<\/h([1|2|3|4|5])\>/", function ( $matches ) {
```

Comment:
There is no guard to prevent this matching a heading which already contains linked text, and a link cannot appear inside a link so one of them would break.